pkg/proxy/proxy.go

```diff
@@ -517,6 +527,43 @@ func Init(cfg *Config) (*Proxy, error) {
 	return p, nil
 }
 
+// ExtractRequestData takes an interface with request data from a Stripe event payload
+// and properly parses it into a StripeRequest struct before returning it
+func ExtractRequestData(data interface{}) (StripeRequest, error) {
+	var req StripeRequest
+
+	reqDataValue := reflect.ValueOf(data)
+	// here we check the type of the RequestData as it could be either a string or a map
+	// this depends on which API version is in use when listening to events
+	switch reqDataValue.Kind() {
+	case reflect.String:
+		req = StripeRequest{
+			ID:             data.(string),
+			IdempotencyKey: "",
+		}
+	case reflect.Map:
+		reqDataMap := reqDataValue.Interface().(map[string]interface{})
+
+		var id = ""
+		if reqDataMap["id"] != nil {
```

Comment:
nit: we've generally been using the two-valued construct for maps in the codebase: https://github.com/stripe/stripe-cli/blob/529cdff8b011fa59fdd872a350b51dfa7e55030e/pkg/fixtures/triggers.go#L112

Reply:
thank you for this callout! Will update